Formula/php.rb

```diff
@@ -0,0 +1,335 @@
+        "INSTALL_IT = $(mkinstalldirs) '#{lib}/httpd/modules' \\2 LIBEXECDIR='#{lib}/httpd/modules' \\4 \\5"
+
+      # Reorder linker flags to put system paths at the end to avoid accidential system linkage
+      %w[EXTRA_LDFLAGS EXTRA_LDFLAGS_PROGRAM].each do |mk_var|
+        system_libs = []
+        other_flags = []
+        s.get_make_var(mk_var).split.each do |f|
+          if f[%r{^-L\/(?:Applications|usr\/lib)\/}]
+            system_libs << f
+            next
+          end
+          other_flags << f
+        end
+        s.change_make_var! mk_var, other_flags.concat(system_libs).join(" ")
+      end
+    end
+
+    # Shared module linker flags come after the sytem flags, prepend to avoid accidential system linkage
+    ENV.prepend "LDFLAGS", "-L#{Formula["tidy-html5"].opt_lib}"
+
+    system "make"
+    ENV.deparallelize
+    system "make", "install"
+
+    bin.install_symlink "phar.phar" => "phar"
+
+    config_path.install "php.ini-development" => "php.ini"
+    (config_path/"php-fpm.d").install "sapi/fpm/www.conf"
+    config_path.install "sapi/fpm/php-fpm.conf"
+    inreplace config_path/"php-fpm.conf", /^;?daemonize\s*=.+$/, "daemonize = no"
+
+    # patch PEAR so it installs extensions outside of the Cellar
+    Dir.chdir prefix do
+      pear_patch = Patch.create :p1, <<-EOS.undent
+        --- a/lib/php/PEAR/Builder.php	2017-09-07 23:46:07.000000000 -0500
+        +++ b/lib/php/PEAR/Builder.php	2017-09-07 23:47:17.000000000 -0500
+        @@ -405,7 +405,7 @@
+                 $prefix = exec($this->config->get('php_prefix')
+                                 . "php-config" .
+                                $this->config->get('php_suffix') . " --prefix");
+        -        $this->_harvestInstDir($prefix, $inst_dir . DIRECTORY_SEPARATOR . $prefix, $built_files);
+        +        $this->_harvestInstDir($this->config->get('ext_dir'), $inst_dir . DIRECTORY_SEPARATOR . $prefix, $built_files);
+                 chdir($old_cwd);
+                 return $built_files;
+             }
+        --- a/lib/php/PEAR/Command/Install.php	2017-09-07 23:45:56.000000000 -0500
+        +++ b/lib/php/PEAR/Command/Install.php	2017-09-07 23:42:22.000000000 -0500
+        @@ -379,7 +379,7 @@
+                     $newini = array();
+                 }
+                 foreach ($binaries as $binary) {
+        -            if ($ini['extension_dir']) {
+        +            if ($ini['extension_dir'] && $ini['extension_dir'] === $this->config->get('ext_dir')) {
+                         $binary = basename($binary);
+                     }
+                     $newini[] = $enable . '="' . $binary . '"' . (OS_UNIX ? "\\n" : "\\r\\n");
+      EOS
+      pear_patch.apply
+    end
+  end
+
+  def caveats
+    s = []
+
+    s << <<-EOS.undent
+      To enable PHP in Apache add the following to httpd.conf and restart Apache:
+          LoadModule php7_module #{HOMEBREW_PREFIX}/lib/httpd/modules/libphp7.so
```

Comment:
Not sure why `HOMEBREW_PREFIX/lib` was preferred over `opt_lib`, but they point to the same thing.

Reply:
@ilovezfs you made that comment, is there a specific reason ?

Reply:
opt_prefix is a symlink to a keg in the Cellar. Presumably we want to be looking for other extensions besides the ones in the Cellar.

Reply:
This configuration line is just for the referenced php7 module. Other module lines would be expected to have their own specific path (most being relative to the httpd opt_prefix)

Reply:
It gives the impression that this is some other location but I don't think it makes a difference one way or the other unless we're trying to make this formula work while not brew linked.

Reply:
Coming back to this I think we need to have this versioned since we want to support multiple php version installed at the same time.